bot/tests/base_test.py

```diff
@@ -0,0 +1,117 @@
+                    self.logger.info(f"Bot response: {message.text}")
+                    return message
+
+            time.sleep(poll_interval)
+
+        raise TimeoutError(msg.bot_response_timeout())
+
+    def assert_response_contains(self, response: Message, expected_fragments: List[str]):
+        error_message = msg.missing_fragment("{fragment}", "{response}")
+        return self.__check_response_fragments(expected_fragments, response, error_message)
+
+    def assert_video_matches(self, response: Message, expected_video_filename: str, received_video_filename: str = 'received_video.mp4'):
+        assert response.media is not None, msg.video_not_returned()
+
+
+        received_video_path = Path(str(self.client.download_media(response, file=received_video_filename)))
+
+        expected_video_path = Path(__file__).parent / 'expected' / 'expected_videos' / expected_video_filename
+
+        assert expected_video_path.exists(), msg.video_not_found(expected_video_filename)
+
+        expected_hash = self.__compute_file_hash(expected_video_path)
+        received_hash = self.__compute_file_hash(received_video_path)
+
+        assert expected_hash == received_hash, msg.video_mismatch()
+
+        received_video_path.unlink()
+        self.logger.info(msg.video_test_success(expected_video_filename))
+
+    def assert_file_matches(self, response: Message, expected_file_filename: str, expected_extension: str, received_file_filename: str = 'received_file'):
+        assert response.media is not None, msg.file_not_returned()
+
+        assert response.file.ext == expected_extension, msg.file_extension_mismatch(expected_extension, response.file.ext)
+
+        received_file_path = Path(str(self.client.download_media(response, file=f'{received_file_filename}{expected_extension}')))
+
+        expected_file_path = Path(__file__).parent / 'expected' / 'expected_files' / expected_file_filename
+
+        assert expected_file_path.exists(), msg.file_not_found(expected_file_filename)
+
+        expected_hash = self.__compute_file_hash(expected_file_path)
+        received_hash = self.__compute_file_hash(received_file_path)
+
+        assert expected_hash == received_hash, msg.file_mismatch()
+
+        received_file_path.unlink()
+        self.logger.info(msg.file_test_success(expected_file_filename))
+
+    @staticmethod
+    def __compute_file_hash(file_path: Path, hash_function: str = 'sha256'):
+        hash_func = hashlib.new(hash_function)
+        with file_path.open('rb') as f:
+            for chunk in iter(lambda: f.read(4096), b''):
+                hash_func.update(chunk)
+        return hash_func.hexdigest()
+
+    @staticmethod
+    def __check_response_fragments(expected_fragments: List[str], response: Message, error_message: str):
+        for fragment in expected_fragments:
+            if fragment not in response.text:
+                raise AssertionError(error_message.format(fragment=fragment, response=response.text))
```

Comment:
holy shit xD po to sa asserty frameworka zebys takiego czegos nie uprawial

Reply:
wgl ale technologia XD 

@pytest.mark.parametrize("x, y, expected", [(1, 2, 3), (2, 3, 5), (3, 5, 8)])
def test_addition(x, y, expected):
    assert x + y == expected

Reply:
akurat co do tego dekoratora to nie powiedzialbym ze jest pomocny w czytelnosci xDDDDDDDDDDD

Reply:
nie wiem na czym stanęło ale doslzliśmy do tego żę assert git, ale takie coś nie XDD